components/date-picker/date-picker.component.ts

```diff
@@ -41,6 +41,8 @@ import { CompatibleDate, DisabledTimeFn, NzDateMode, PresetRanges, SupportTimeOp
 const POPUP_STYLE_PATCH = { position: 'relative' }; // Aim to override antd's style to support overlay's position strategy (position:absolute will cause it not working beacuse the overlay can't get the height/width of it's content)
 const NZ_CONFIG_MODULE_NAME: NzConfigKey = 'datePicker';
 
+export type NzDatePickerType = 'large' | 'default' | 'small';
```

Comment:
Thank you for the PR, this type need to export in `date-picker/public-api.ts`

Reply:
Use `NzDatePickerSizeType` would be better.